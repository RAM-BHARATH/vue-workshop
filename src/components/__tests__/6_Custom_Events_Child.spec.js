import Vue from "vue";
import Vuetify from "vuetify";
import { mount, createLocalVue } from "@vue/test-utils";

Vue.use(Vuetify);
const localVue = createLocalVue();

import Child from "../6_Custom_Events_Child";

describe("6_Custom_Events_Child", () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const getWrapper = () =>
    mount(Child, {
      localVue,
      vuetify
    });

  it("should render child component", () => {
    const wrapper = getWrapper();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should emit event when closing snackbar", async () => {
    const wrapper = getWrapper();
    const event = jest.fn();
    const button = wrapper.find(".v-btn");

    wrapper.vm.$on("unlock-button", event);
    expect(event).toHaveBeenCalledTimes(0);
    button.trigger("click");
    expect(event).toHaveBeenCalledTimes(1);
  });
});
