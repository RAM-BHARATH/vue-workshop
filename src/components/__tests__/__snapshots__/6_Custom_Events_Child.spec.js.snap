// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`6_Custom_Events_Child should render child component 1`] = `
<div class="v-snack v-snack--active v-snack--bottom v-snack--has-background" style="padding: 0px 0px 0px 0px;">
  <transition-stub name="v-snack-transition">
    <div class="v-snack__wrapper v-sheet theme--dark">
      <div role="status" aria-live="polite" class="v-snack__content">
        Hey, when you close me button will unlock ðŸ˜‰
      </div>
      <div class="v-snack__action "><button type="button" class="v-btn v-btn--flat v-btn--text theme--dark v-size--default red--text v-snack__btn"><span class="v-btn__content">
      Close
    </span></button></div>
    </div>
  </transition-stub>
</div>
`;
